nagiosData.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func nagiosData(cmdFactory getNagiosDataCmdFactory, project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		stdOut, stdOutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdOutCloser.Close()
+
+		stdErr, stdErrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdErrCloser.Close()
+
+		stdOutBuff := bytes.NewBuffer([]byte{})
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			stdErr.Write([]byte(err.Error()))
+			return err
+		}
+		cmd := cmdFactory(project, nagiosPod[0])
+
+		if err := runCmdCaptureOutput(cmd, stdOutBuff, stdErr); err != nil {
+			stdErr.Write([]byte(err.Error()))
+			return err
+		}
+
+		stdOut.Write(parseNagiosStatus(string(stdOutBuff.Bytes())))
```

Comment:
@philbrookes io.MultiWriter(parseNagiosStatus(string(stdOutBuff.Bytes())), stdOut) or even much more readable. Here nobody knows what you read from and what you write to StdOut